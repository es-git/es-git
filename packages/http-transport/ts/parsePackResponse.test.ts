import parsePackResponse from './parsePackResponse';
import streamToAsyncIterator from './utils/streamToAsyncIterator';
import * as fs from 'fs';

const sampleFile = __dirname + '/../samples/fetchResponse.txt';

test('parse side-band-64', async () => {
  const expectedTypes = [
    'nak',
    'progress',
    'progress',
    'progress',
    'progress',
    'progress',
    'progress',
    'progress',
    'pack',
    'pack',
    'progress',
  ];
  const actualTypes = [];
  let actualProgress = '';
  for await(const result of parsePackResponse(streamToAsyncIterator(fs.createReadStream(sampleFile)))){
    actualTypes.push(result.type);
    if(result.type === 'progress'){
      actualProgress += result.message;
    }
  }
  expect(actualTypes).toEqual(expectedTypes);
  expect(actualProgress).toMatchSnapshot();
});
