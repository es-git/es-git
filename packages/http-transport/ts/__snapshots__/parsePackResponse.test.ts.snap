// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse side-band-64 1`] = `
"Counting objects: 45, done.
Compressing objects:  11% (1/9)   
Compressing objects:  22% (2/9)   
Compressing objects:  33% (3/9)   
Compressing objects:  44% (4/9)   
Compressing objects:  55% (5/9)   
Compressing objects:  66% (6/9)   
Compressing objects:  77% (7/9)   
Compressing objects:  88% (8/9)   
Compressing objects: 100% (9/9)   
Compressing objects: 100% (9/9), done.
Total 45 (delta 2), reused 7 (delta 0), pack-reused 36
"
`;
